nemo/collections/nlp/models/language_modeling/transformer_lm_model.py

```diff
@@ -142,145 +121,155 @@ def training_step(self, batch, batch_idx):
-                'file_name' in cfg and cfg['file_name'] is None
-            ):
-                logging.warning(
-                    "Could not load dataset as `file_name` was None or "
-                    f"`tarred_text_filepaths` is None. Provided config : {cfg}"
-                )
-                return None
-
-            shuffle_n = cfg.get('shuffle_n', 4 * cfg['batch_size']) if shuffle else 0
-            dataset = TarredL2RLanguageModelingDataset(
-                text_tar_filepaths=cfg['tarred_text_filepaths'],
-                metadata_path=cfg['file_name'],
+
+        if cfg.get("use_tarred_dataset", False):
+            if cfg.get("metadata_file") is None:
+                raise FileNotFoundError("Trying to use tarred data set but could not find metadata path in config.")
+            else:
+                metadata_file = cfg.get('metadata_file')
+                with open(metadata_file) as metadata_reader:
+                    metadata = json.load(metadata_reader)
+                if cfg.get('tar_files') is None:
+                    tar_files = metadata.get('tar_files')
+                    if tar_files is not None:
+                        logging.info(f'Loading from tarred dataset {tar_files}')
+                    else:
+                        raise FileNotFoundError("Could not find tarred dataset in config or metadata.")
+                else:
+                    tar_files = cfg.get('tar_files')
+                    if metadata.get('tar_files') is not None:
+                        raise ValueError(
+                            'Tar files specified in config and in metadata file. Tar files should only be specified once.'
+                        )
+            dataset = TarredOneSideTranslationDataset(
+                text_tar_filepaths=tar_files,
+                metadata_path=metadata_file,
                 tokenizer=self.tokenizer,
-                max_seq_length=self.dataset_cfg.max_seq_length,
-                batch_step=predict_last_k,
-                shuffle_n=shuffle_n,
-                shard_strategy=cfg.get("tarred_shard_strategy", "scatter"),
+                shuffle_n=cfg.get("tar_shuffle_n", 100),
+                shard_strategy=cfg.get("shard_strategy", "scatter"),
                 global_rank=self.global_rank,
                 world_size=self.world_size,
             )
-
-            shuffle = False
+            return torch.utils.data.DataLoader(
+                dataset=dataset,
+                batch_size=1,
+                num_workers=cfg.get("num_workers", 2),
+                pin_memory=cfg.get("pin_memory", False),
+                drop_last=cfg.get("drop_last", False),
+            )
         else:
-            if "file_name" in cfg and cfg.file_name is None:
-                logging.warning(f"Could not load dataset as `file_name` was None. Provided config : {cfg}")
-                return None
-
-            dataset = L2RLanguageModelingDataset(
+            dataset = TranslationOneSideDataset(
```

Comment:
Since this is now being used for both LM and Translation, maybe it should have a more generic name? `SentenceLevelDataset` maybe?

Reply:
Yep, I was also thinking about that. Renamed OneSideTranslationDataset to SentenceDataset and moved it into nlp/data/language_modeling.